File: algorithm/HDU-ACM/ambition0109/1568.cpp
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2010-09-21 17:06:17
////Compiler: Visual C++
//////////////////////////////////////////////////////
////Problem ID: 1568
////Problem Title: 
////Run result: Accept
////Run time:0MS
////Run memory:224KB
//////////////////System Comment End//////////////////
#include<cstdio>
#include<cmath>
using namespace std;

int Fib[]={0,1,1,2,3,5,8,13,21,34,55,89,144,233,377,610,987,1597,2584,4181,6765};
int main()
{
	int n;
	while(scanf("%d",&n)!=EOF){
		if(n<21){
			printf("%d\n",Fib[n]);
		}else{
			double ffour=-0.5*(log10(5.0))+n*log10((sqrt(5.0)+1.0)/2.0);
            ffour=pow(10.0,double(ffour-(int)ffour));
            while(ffour<1000) ffour*=10;
            printf("%d\n",int(ffour));
		}
	}
}

File: algorithm/HDU-ACM/ambition0109/1568.java
////////////////////System Comment////////////////////
////Welcome to Hangzhou Dianzi University Online Judge
////http://acm.hdu.edu.cn
//////////////////////////////////////////////////////
////Username: ambition0109
////Nickname: Ambition
////Run ID: 
////Submit time: 2011-04-11 22:20:42
////Compiler: Java
//////////////////////////////////////////////////////
////Problem ID: 1568
////Problem Title: 
////Run result: Accept
////Run time:171MS
////Run memory:7144KB
//////////////////System Comment End//////////////////
import java.util.Scanner;

public class Main {
	static int[] Fib=new int[]{0,1,1,2,3,5,8,13,21,34,55,89,144,233,377,610,987,1597,2584,4181,6765};
	static public void main(String args[]){
		Scanner in=new Scanner(System.in);
		while(in.hasNext()){
			int n=in.nextInt();
			if(n<21){
				System.out.println(Fib[n]);
			}else{
				double ff=-0.5*(Math.log10(5.0))+n*Math.log10((Math.sqrt(5.0)+1.0)/2.0);
				ff=Math.pow(10,(double)(ff-(int)ff));
				while(ff<1000) ff*=10;
				System.out.println((int)ff);
			}
		}
	} 
}
